pyproject.toml

```
@@ -0,0 +1,270 @@
+[project]
+name            = "gigl"
+description     = "GIgantic Graph Learning Library"
+readme          = "README.md"
+version         = "0.1.0"
+classifiers     = [
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.11",
+]
+requires-python = "==3.11.*" # Limited by tfx-bsl not having wheels available for newer python versions.
+
+dependencies = [
+  "argo-workflows",
+  "chardet",
+  "google-cloud-aiplatform",
+  "google-cloud-dataproc",
+  "google-cloud-logging",
+  "google-cloud-pipeline-components",
+  "google-cloud-storage",
+  # ipython and ipykernel are required for us to run jupyter notebooks as a part of our test suite.
+  # See https://stackoverflow.com/questions/69759351/error-jupyter-client-kernelspec-nosuchkernel-no-such-kernel-named-python3-occu
+  "ipykernel",
+  "ipython",
+  "kfp>=2.0.0",
+  "matplotlib",
+  "mmh3",
+  "msgpack",
+  "nbconvert",
+  "nbformat",
+  "numpy",
+  "omegaconf>=2.3.0,<3.0.0",
+  "pandas",
+  "protobuf",
+  "python-dotenv",
+  "pyyaml",
+  "scikit-learn",
+  "pip~=25.3", # Used by KFP Python components - need to install pip as a package
```

```suggestion
dependencies = [
  "argo-workflows",
  "chardet",
  "google-cloud-aiplatform",
  "google-cloud-dataproc",
  "google-cloud-logging",
  "google-cloud-pipeline-components",
  "google-cloud-storage",
  # ipython and ipykernel are required for us to run jupyter notebooks as a part of our test suite.
  # See https://stackoverflow.com/questions/69759351/error-jupyter-client-kernelspec-nosuchkernel-no-such-kernel-named-python3-occu
  "ipykernel",
  "ipython",
  "kfp>=2.0.0",
  "matplotlib",
  "mmh3",
  "msgpack",
  "nbconvert",
  "nbformat",
  "numpy",
  "omegaconf>=2.3.0,<3.0.0",
  "pandas",
  "pip~=25.3", # Used by KFP Python components - need to install pip as a package
  "protobuf",
  "python-dotenv",
  "pyyaml",
  "scikit-learn",
```

nit. shall we try to keep this sorted?